app/frontend/src/features/profile/view/leaveReference/ReferenceDataContext.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { createContext, ReactNode, useContext, useState } from "react";
+
+export type ReferenceContextFormData = {
+  text: string;
+  wasAppropriate: string;
+  rating: number;
+};
+
+interface ReferenceDataProviderProps {
+  children: ReactNode;
+}
+
+type ReferenceContextType = {
+  data: ReferenceContextFormData;
+  setValues: (data: ReferenceContextFormData) => void;
+};
+
+const ReferenceDataContext = createContext<ReferenceContextType | undefined>(
+  undefined
+);
+
+export const ReferenceDataProvider = ({
+  children,
+}: ReferenceDataProviderProps) => {
+  const [data, setData] = useState({
+    text: "",
+    wasAppropriate: "",
+    rating: 0,
+  });
+
+  const setValues = (values: ReferenceContextFormData) => {
+    setData((prevData) => ({
+      ...prevData,
+      ...values,
+    }));
+  };
+
+  return (
+    <ReferenceDataContext.Provider value={{ data, setValues }}>
+      {children}
+    </ReferenceDataContext.Provider>
+  );
+};
+
+export const useReferenceData = () => useContext(ReferenceDataContext);
```

Comment:
So while I've built multi-step forms in React without the use of a Context, I can't find any documentation nor implementations of a multi-step from using `react-hook-form` that doesn't use either `Context`or a state management library (most common). Neither can I get this working as a multi step form by removing the Context and passing `data` and `setValues` from the `ReferenceForm` component. That being said, I can't find anything explicitly saying that doing it this way is impossible, so I'm a bit lost as to what to do – it may just be my relative inexperience? We have used `useForm` without context of course, but again it was not a multi step form like this.

Reply:
In the component that contains all the steps, just have a normal local state with `useState` and pass `referenceData` and `setReferenceData` to each step component via props